--- jack-capture-0.9.57.orig/jack_capture.c
+++ jack-capture-0.9.57/jack_capture.c
@@ -129,6 +129,8 @@ static bool jack_has_been_shut_down=fals
 static int64_t unreported_overruns=0;
 static int total_overruns=0;
 
+volatile int freewheel_mode=0;
+
 
 /* Disk thread */
 #if HAVE_LAME
@@ -1505,7 +1507,13 @@ static void stop_connection_thread(void)
 }
 
 static int graphordercallback(void *arg){
-  wake_up_connection_thread();
+  if (!freewheel_mode)
+    wake_up_connection_thread();
+  return 0;
+}
+
+static int freewheelcallback(int starting, void *arg){
+  freewheel_mode = starting;
   return 0;
 }
 
@@ -1825,6 +1833,8 @@ void init_various(void){
 
     jack_set_graph_order_callback(client,graphordercallback,NULL);
 
+    jack_set_freewheel_callback(client,freewheelcallback,NULL);
+
     if (jack_activate(client)) {
       fprintf (stderr,"\nCan not activate client");
       exit(-2);
